simul/platform/runsimul.go

```diff
@@ -55,11 +61,19 @@ func Simulate(serverAddress, simul, monitorAddress string) error {
 		if err != nil {
 			return err
 		}
-		err = sim.Node(sc)
-		if err != nil {
-			return err
-		}
 		sims[i] = sim
+		func(sc *onet.SimulationConfig) {
+			server.RegisterProcessorFunc(simulInitID, func(env *network.Envelope) {
+				err = sim.Node(sc)
+				if err != nil {
+					log.Error(err)
+				}
+				sc.Server.Send(env.ServerIdentity, &simulInitDone{})
+			})
+			server.RegisterProcessorFunc(simulInitDoneID, func(env *network.Envelope) {
+				wgSimulInit.Done()
+			})
+		}(sc)
```

Comment:
I did some tests in go-playground which seemed to show that I cannot rely on `sc` being correctly passed to the `ProcessorFunc`, except if I pass it as an argument to a function.